sql: clean up comments in legacy DROP TYPE

rafiss opened the pull request:
The cleanup logic was not removing comments on types. This was caught by the new validation added in #146213.

This patch adds a logic test that reveals the bug, and also adds validation to TestWorkload. This validation is already in the schemachange/random-load roachtest. Doing it here too can help catch bugs earlier.

fixes https://github.com/cockroachdb/cockroach/issues/146516
fixes https://github.com/cockroachdb/cockroach/issues/146913
fixes https://github.com/cockroachdb/cockroach/issues/146793

Release note: None

rafiss committed:
sql: clean up comments in legacy DROP TYPE

The cleanup logic was not removing comments on types.

This patch adds a logic test that reveals the bug, and also adds
validation to TestWorkload. This validation is already in the
schemachange/random-load roachtest. Doing it here too can help
catch bugs earlier.

Release note: None

Dedej-Bergin reviewed (commented):
Nice change! I just have a couple of minor questions.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/147173#-:-OQtUyRC6XWiXBwCkWSw:b-tmyztw)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained

___
*[`pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go` line 75 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/147173#-OQtWjg9Bg1TcE2QFPYB:-OQtWjg9Bg1TcE2QFPYC:b-pmlpat) ([raw file](https://github.com/cockroachdb/cockroach/blob/9188ab784d55c70fd4ffa09d7f1f035245b73722/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go#L75)):*
> ```Go
> 	}
> 
> 	// Reusable validation function.
> ```

I see the schema_change_external_test.go file as our main file, it could be better to keep it smaller. Would it make sense to move helper methods like this to another file, maybe to `pkg/workload/schemachange/error_screening.go`?
___
*[`pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go` line 138 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/147173#-OQtY1ZqCd1ylbkKKKoT:-OQtY1ZqCd1ylbkKKKoU:b-lk8d4i) ([raw file](https://github.com/cockroachdb/cockroach/blob/9188ab784d55c70fd4ffa09d7f1f035245b73722/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go#L138)):*
> ```Go
> 		// Drop the database and run validation again. Test DROP DATABASE behavior
> 		// with legacy schema changer 50% of the time.
> 		schemaChangerSetting := "on"
> ```

I see the schema_change_external_test.go file as our main file, it could be better to keep it smaller. Do you think it could be good to move this `DROP DATABASE CASCADE` test to `pkg/workload/schemachange/operation_generator.go` and call it from there?


<!-- Sent from Reviewable.io -->

rafiss committed:
workload/schemachange: refactor invalid object validation into shared helper

Extract duplicate validation logic from schemachange tests into a shared
ValidateInvalidObjects function in the workload package.

Release note: none

rafiss reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/147173)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @Dedej-Bergin, @golgeek, and @herkolategan)

___
*[`pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go` line 75 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/147173#-OQtWjg9Bg1TcE2QFPYB:-OR4_5120tqnrx-yjn5x:b-l72f7b) ([raw file](https://github.com/cockroachdb/cockroach/blob/9188ab784d55c70fd4ffa09d7f1f035245b73722/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go#L75)):*
<details><summary><i>Previously, Dedej-Bergin (Bergin Dedej) wrote…</i></summary><blockquote>

I see the schema_change_external_test.go file as our main file, it could be better to keep it smaller. Would it make sense to move helper methods like this to another file, maybe to `pkg/workload/schemachange/error_screening.go`?
</blockquote></details>

i moved some of the validation logic into a new file, and made the schemachange roachtest use it also.
___
*[`pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go` line 138 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/147173#-OQtY1ZqCd1ylbkKKKoT:-OR4h12U8t2EQY6xSmUn:bvkyp1e) ([raw file](https://github.com/cockroachdb/cockroach/blob/9188ab784d55c70fd4ffa09d7f1f035245b73722/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go#L138)):*
<details><summary><i>Previously, Dedej-Bergin (Bergin Dedej) wrote…</i></summary><blockquote>

I see the schema_change_external_test.go file as our main file, it could be better to keep it smaller. Do you think it could be good to move this `DROP DATABASE CASCADE` test to `pkg/workload/schemachange/operation_generator.go` and call it from there?
</blockquote></details>

i think this one is a small enough code snippet that moving it to another package would decrease code legibility


<!-- Sent from Reviewable.io -->

Dedej-Bergin reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/147173#-:-ORBoD3a99z9CKajodop:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @golgeek and @herkolategan)



<!-- Sent from Reviewable.io -->

rafiss commented:
tftr!

bors r+
